### PR TITLE
Improve GitHub pack with support for managing releases and deployments (+ add action aliases)

### PR DESCRIPTION
**Pull request by jjm**
### Status 

- pack extension.

### Description

Add actions using the GitHub APIs directly so that it's possible to manage GitHub releases & deployments via ST2 (and via ChatOPs).

### Checklist (tick everything that applies)

- [X] Version bump (required for changed packs)
- [X] Code linting (required, can be done after the PR checks)
- [X] [Tests](https://docs.stackstorm.com/development/pack_testing.html) (not required but really recommended)
- [X] Add Action Alias tests
- [x] Add support for connecting to GitHub.com and Enterprise at the same time.

### CODE REVIEW

File: packs/github/config.schema.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+token:
+  description: "GitHub oAuth Token"
+  type: "string"
+  default: "{{user.github_enterprise_oauth_token}}"
```
Comment (Kami) [inline packs/github/config.schema.yaml]: Nice, thanks!

File: packs/github/config.schema.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+user: 
+  description: "GitHub Username"
+  type: "string"
+
+password:
+  description: "GitHub Password"
+  type: "string"
+
+default:
```
Comment (Kami) [inline packs/github/config.schema.yaml]: Maybe `user_type` or `account_type` is a better name for this config option?

Comment (Kami) [inline packs/github/config.schema.yaml]: And it looks like in another place `github_type` parameter name is used so it's also good to be consistent and use the same name across the board :)

File: packs/github/actions/create_deployment_status.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import datetime
+
+from lib.base import BaseGithubAction
+
+
+class CreateDeploymentAction(BaseGithubAction):
+    def run(self, api_user, repository, deployment_id, state,
+            description, github_type):
+
+        valid_states = ["pending", "success", "error", "failure"]
```
Comment (Kami) [inline packs/github/actions/create_deployment_status.py]: Minor consistency thing - we tend to use use single quotes around strings.

File: packs/github/config.schema.yaml (hunk 1 above)
Comment (jjm) [inline packs/github/config.schema.yaml]: Turns out I got this totally wrong :-( and forgot to correct it...

I've changed it to `system.github_oauth_token` as `user` scoping does not apply to action aliases.

File: packs/github/config.schema.yaml (hunk 2 above)
Comment (jjm) [inline packs/github/config.schema.yaml]: Cool catch, I'll change it to `github_type` and only have _default_ in the description.

File: packs/github/actions/create_deployment_status.py (hunk 3 above)
Comment (jjm) [inline packs/github/actions/create_deployment_status.py]: Cheers, fixed!

**Comment by jjm** [comment]
Thanks for the review comments @Kami, these have been addressed and it's now ready to be merged if it looks ok to you too :-D

File: packs/github/config.schema.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  type: "string"
+
+password:
+  description: "GitHub Password"
+  type: "string"
+
+github_type:
+  description: "Default to either github or enterprise."
+  type: "string"
+  default: "online"
```
Comment (Kami) [inline packs/github/config.schema.yaml]: Maybe `public` or similar would be better than `online` but not a big deal anyway.

**Comment by Kami** [comment]
Merged into master.

Nice works and thanks! :+1:

**Pull request merged**